### ISSUE TITLE
Enforcing attribute naming by type

### ISSUE THREAD

**Issue by mfridman** [go-simpler/sloglint#59]
What do you think about adding the ability to enforce naming for certain types?

One that comes to mind is enforcing error naming. E.g., if the attribute is an `error` then it must be named `"error"` or some user-configurable list like:

```
error-type-names: ["error", "secondary_error"]
```

**Comment by mattdowdell** [comment]
In slog, kinds (value types) do not treat errors as a first class citizen. Instead, you have to convert them to strings first. Did you have a potential implementation in mind for this feature?

**Comment by tmzane** [comment]
> One that comes to mind is enforcing error naming. E.g., if the attribute is an error then it must be named "error" or some user-configurable list like:

@mfridman I agree that the use case with naming the error key is legit (happened to me), but I can't think of anything similar for other types. Most fields are just ints and strings, so forbidding the use of keys other than those defined in the linter config for all, e.g. `slog.Int` calls, would mean that you'd have to update the config for every new key, even for a simple debug log. Doesn't sound very convenient to me. I usually define constants for keys and forbid the use of literal keys in log calls, I find it kinda similar to what you want. Wouldn't that work for you? We have the `no-raw-keys` for exactly that.

**Comment by mfridman** [comment]
Yeah, I've settled on using raw literals, and it's good enough.

Closing this issue. Thanks for the response.
